### PR TITLE
[ISSUE 4966]Optimize ThreadLocalIndex and remove redundant else

### PR DESCRIPTION
**Pull request by 123monkey**
**Make sure to set the target branch to `develop`**

## What is the purpose of the change
This is to fix https://github.com/apache/rocketmq/issues/4966
remove redundant else

## Brief changelog

remove redundant else

## Verifying this change

remove redundant else

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by Oliverwqcwrw — APPROVED** [review]

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/TopicPublishInfo.java [hunk 1]
```diff
@@ -69,19 +69,18 @@ public void setHaveTopicRouterInfo(boolean haveTopicRouterInfo) {
...
-                MessageQueue mq = this.messageQueueList.get(pos);
-                if (!mq.getBrokerName().equals(lastBrokerName)) {
-                    return mq;
-                }
+        } 
+        for (int i = 0; i < this.messageQueueList.size(); i++) {
+            int index = this.sendWhichQueue.incrementAndGet();
+            int pos = Math.abs(index) % this.messageQueueList.size();
+            if (pos < 0)
+                pos = 0;
```
Comment (aaron-ai) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/TopicPublishInfo.java:77]:
```suggestion
            if (pos < 0) {
                pos = 0;
            }
```

**Review by aaron-ai — CHANGES_REQUESTED** [review]

**Review by guyinyou — APPROVED** [review]
lgtm

**Review by tsunghanjacktsai — DISMISSED** [review]

**Comment by lizhanhui** [comment]
@123monkey Can you please rebase on top of the latest `develop` branch?

**Review by ni-ze — APPROVED** [review]
LGTM

**Review by aaron-ai — APPROVED** [review]

**Review by tsunghanjacktsai — APPROVED** [review]

**Review by Oliverwqcwrw — APPROVED** [review]

**Review by caigy — APPROVED** [review]

**Review by guyinyou — APPROVED** [review]

**Pull request closed without merging**